rj_gameplay/rj_gameplay/play/penalty_defense.py

```diff
@@ -1,137 +1,75 @@
-        flat_results = self.role_assigner.assign_roles(
-            flat_requests, world_state, prev_results
-        )
-        role_results = play.unflatten_results(flat_results)
+from stp.role.assignment.naive import NaiveRoleAssignment
 
-        # Get list of all SkillEntries from all tactics
-        skills = []
-        for tactic in self.tactics:
-            skills += tactic.tick(world_state, role_results[tactic])
+from rj_gameplay.tactic import line_tactic, goalie_tactic
 
-        # Get all role assignments
-        # SkillEntry to (list of?) RoleResult
-        skill_dict = {}
-        for tactic in self.tactics:
-            skill_dict.update(role_results[tactic])
+from typing import List, Tuple
+from enum import Enum, auto
 
-        return (skill_dict, skills)
 
-    def is_done(self, world_state):
-        # returns done when all tactics are done
-        # TODO: change all is_done() to use all()?
-        return all([tactic.is_done(world_state) for tactic in self.tactics])
+class State(Enum):
+    INIT = auto()
+    ACTIVE = auto()
+    DONE = auto()
 
 
-class PenaltyDefense(play.IPlay):
-    """For when we don't have the ball and are trying to stop the opponent from scoring."""
+class PenaltyDefense(stp.play.Play):
+    """
+    Play consistas of:
+                    -5 wallers
+                    -1 goalie
+    The goalie will defend the opponents penalty kick.
+    """
 
     def __init__(self):
-        self.tactics = [
-            goalie_tactic.GoalieTactic(),
-            move_tactic.Move((1.6, 9.0)),
-            move_tactic.Move((1.9, 9.0)),
-            move_tactic.Move((2.2, 9.0)),
-            move_tactic.Move((2.5, 9.0)),
-            move_tactic.Move((2.8, 9.0)),
-        ]
-
-        self.role_assigner = NaiveRoleAssignment()
-
-    def compute_props(self, prev_props):
-        pass
+        super().__init__()
+        self._state = State.INIT
 
     def tick(
         self,
-        world_state: rc.WorldState,
-        prev_results: role.assignment.FlatRoleResults,
-        props,
-    ) -> Tuple[
-        Dict[Type[tactic.SkillEntry], List[role.RoleRequest]],
-        List[tactic.SkillEntry],
-    ]:
-
-        # Get role requests from all tactics and put them into a dictionary
-        role_requests: play.RoleRequests = {
-            tactic: tactic.get_requests(world_state, None) for tactic in self.tactics
-        }
-
-        # Flatten requests and use role assigner on them
-        flat_requests = play.flatten_requests(role_requests)
-        flat_results = self.role_assigner.assign_roles(
-            flat_requests, world_state, prev_results
-        )
-        role_results = play.unflatten_results(flat_results)
+        world_state: stp.rc.WorldState,
+    ) -> List[RobotIntent]:
+        if self._state == State.INIT:
+            self.prioritized_tactics.append(goalie_tactic.GoalieTactic(world_state, 0))
+            self.prioritized_tactics.append(line_tactic.LineTactic(world_state, 5))
```

Comment:
Looking at line_tactic, I don't actually think how its currently written would be legal. The robots that aren't the goalie have to be 1 meter behind the ball, and I think line_tactic lines them up along the left middle. You could try to change line_tactic to take in a set of points or something so you have control over where the robots line up, or you could use 5 move commands. 
See penalty rules in section 5.3.5 here: https://robocup-ssl.github.io/ssl-rules/sslrules.pdf

Reply:
Hey, just adding context: the line tactic should be the one that lines up our robots properly for penalty offense/defense. By the rules (in the section Hussain cited): "3. Throughout the penalty kick procedure, all other robots have to be 1m behind the ball such that they do not interfere the penalty kick procedure."

The simple solution is to just move the line to the bottom left/right instead of the middle where it currently is. I will make a new task for this on the board.